internal/ls/findallreferences.go

```diff
@@ -0,0 +1,1490 @@
+		return getAllReferencesForKeyword(
+			sourceFiles,
+			node.Kind,
+			// cancellationToken,
+			node.Kind == ast.KindReadonlyKeyword,
+		)
+	}
+
+	if ast.IsImportMeta(node.Parent) && node.Parent.Name() == node {
+		// !!! unimplemented
+		return nil // getAllReferencesForImportMeta(sourceFiles /*, cancellationToken*/)
+	}
+
+	if node.Kind == ast.KindStaticKeyword && node.Parent.Kind == ast.KindClassStaticBlockDeclaration {
+		return []*SymbolAndEntries{{definition: &Definition{Kind: definitionKindKeyword, node: node}, references: []*Entry{NewNodeEntry(node)}}}
+	}
+
+	// Labels
+	if isJumpStatementTarget(node) {
+		// if we have a label definition, look within its statement for references, if not, then
+		// the label is undefined and we have no results..
+		if labelDefinition := getTargetLabel(node.Parent, node.Text()); labelDefinition != nil {
+			return getLabelReferencesInNode(labelDefinition.Parent, labelDefinition)
+		}
+		return nil
+	}
+
+	if isLabelOfLabeledStatement(node) {
+		// it is a label definition and not a target, search within the parent labeledStatement
+		return getLabelReferencesInNode(node.Parent, node.AsIdentifier())
+	}
+
+	if isThis(node) {
+		return getReferencesForThisKeyword(node, sourceFiles /*, cancellationToken*/)
+	}
+
+	if node.Kind == ast.KindSuperKeyword {
+		return getReferencesForSuperKeyword(node)
+	}
+
+	return nil
+}
+
+func getLabelReferencesInNode(container *ast.Node, targetLabel *ast.Identifier) []*SymbolAndEntries {
+	sourceFile := ast.GetSourceFileOfNode(container)
+	labelName := targetLabel.Text
+	references := core.MapNonNil(getPossibleSymbolReferenceNodes(sourceFile, labelName, container), func(node *ast.Node) *Entry {
+		// Only pick labels that are either the target label, or have a target that is the target label
+		if node == targetLabel.AsNode() || (isJumpStatementTarget(node) && getTargetLabel(node, labelName) == targetLabel) {
+			return NewNodeEntry(node)
+		}
+		return nil
+	})
+	return []*SymbolAndEntries{NewSymbolAndEntries(definitionKindLabel, targetLabel.AsNode(), nil, references)}
+}
+
+func getReferencesForThisKeyword(thisOrSuperKeyword *ast.Node, sourceFiles []*ast.SourceFile) []*SymbolAndEntries {
+	searchSpaceNode := ast.GetThisContainer(thisOrSuperKeyword, false /*includeArrowFunctions*/, false /*includeClassComputedPropertyName*/)
+
+	// Whether 'this' occurs in a static context within a class.
+	staticFlag := ast.ModifierFlagsStatic
+	isParameterName := func(node *ast.Node) bool {
+		return node.Kind == ast.KindIdentifier && node.Parent.Kind == ast.KindParameter && node.Parent.Name() == node
+	}
+
+	// !!! check fall throughs
+	switch searchSpaceNode.Kind {
+	case ast.KindMethodDeclaration, ast.KindMethodSignature:
+		if ast.IsObjectLiteralMethod(searchSpaceNode) {
+			staticFlag &= searchSpaceNode.ModifierFlags()
+			searchSpaceNode = searchSpaceNode.Parent // re-assign to be the owning object literals
+			break
+		}
+		// falls through
```

Comment:
Is the comment here because you're not sure if the original code should be falling through?

Reply:
yup, was a reminder, I've now checked 👍